smelt.lic

```diff
@@ -25,80 +17,76 @@ class Smelt
 
     args = parse_args(arg_definitions)
 
-    get_crafting_item(@rod, @bag, @bag_items, @belt)
+    @settings = get_settings
+    @bag = @settings.crafting_container
+    @bag_items = @settings.crafting_items_in_container
+    @belt = @settings.forging_belt
+    @rod = @settings.forging_tools.find { |item| /rod/ =~ item }
+    @adjustable_tongs = @settings.adjustable_tongs
+
+    Flags.add('ingot-done', 'At last the metal appears to be thoroughly mixed')
+
+    DRCC.get_crafting_item(@rod, @bag, @bag_items, @belt)
     if args.refine
-      refine
+      DRCC.get_crafting_item('flux', @bag, @bag_items, @belt)
+      work("pour my flux in crucible", "flux", true)
     else
-      stir
-    end
-    stow_crafting_item(@rod, @bag, @belt)
-  end
-
-  def refine
-    bput('get flux', 'you get')
-    case bput('pour flux in crucible', 'clumps of molten metal', 'flickers and is unable to consume', 'down and needs more fuel', 'roundtime')
-    when /roundtime/i
-      waitrt?
-      fput('stow flux')
-      return stir
-    when 'clumps of molten metal'
-      waitrt?
-      fput('stow flux')
-      return turn
-    when 'flickers and is unable to consume'
-      waitrt?
-      fput('stow flux')
-      return bellows
-    when 'down and needs more fuel'
-      waitrt?
-      fput('stow flux')
-      return fuel
+      work("stir crucible with my rod", "rod", false)
     end
+    DRCC.stow_crafting_item(@rod, @bag, @belt)
   end
 
-  def stir
-    pause 1
-    waitrt?
-    case bput("stir crucible with my #{@rod}", 'You can only mix a crucible', 'clumps of molten metal', 'flickers and is unable to consume', 'needs more fuel', 'needs some more fuel', 'roundtime')
-    when /roundtime/i
-      return stir
-    when 'You can only mix a crucible'
-      return
-    when 'clumps of molten metal'
-      return turn
-    when 'flickers and is unable to consume'
-      return bellows
-    when 'needs more fuel', 'needs some more fuel'
-      return fuel
+  def work(command, item, stow)
+    loop do
+      break if Flags['ingot-done']
+      case DRC.bput(command, 'Pour what', 'You can only mix a crucible', 'clumps of molten metal', 'flickers and is unable to consume', 'needs more fuel', 'needs some more fuel', 'roundtime')
+      when 'Pour what'
+        DRC.message("Missing Flux")
+        break
+      when 'You can only mix a crucible'
+        break
+      when 'clumps of molten metal'
+        DRCC.stow_crafting_item(@item, @bag, @belt) if DRCI.in_left_hand?(@item)
+        @command = "turn crucible with my rod"
+        @item = "rod"
+        @stow = false
```

Comment:
Are `stow` and `@stow` used any more? I can only find assignments, not usages.

Reply:
no, they're gone.